pkg/controller/containerlaunchpriority/container_launch_priority.go

```diff
@@ -0,0 +1,217 @@
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r *ReconcileContainerLaunchPriority) error {
+	// Create a new controller
+	c, err := controller.New("container-launch-priority-controller", mgr, controller.Options{Reconciler: r, MaxConcurrentReconciles: concurrentReconciles})
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(&source.Kind{Type: &v1.Pod{}}, &handler.EnqueueRequestForObject{}, predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			pod := e.Object.(*v1.Pod)
+			return r.validate(pod)
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			pod := e.ObjectNew.(*v1.Pod)
+			return r.validate(pod) && !r.getConditionStatus(pod, v1.ContainersReady)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return false
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileContainerLaunchPriority{}
+
+// ReconcileContainerLaunchPriority reconciles a Pod object
+type ReconcileContainerLaunchPriority struct {
+	client.Client
+	recorder record.EventRecorder
+}
+
+// +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=pods/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+
+func (r *ReconcileContainerLaunchPriority) Reconcile(request reconcile.Request) (res reconcile.Result, err error) {
+	start := time.Now()
+	if klog.V(3) {
+		klog.Infof("Starting to process Pod %v", request.NamespacedName)
+	}
+	defer func() {
+		if err != nil {
+			klog.Warningf("Failed to process Pod %v, elapsedTime %v, error: %v", request.NamespacedName, time.Since(start), err)
+		} else {
+			klog.Infof("Finish to process Pod %v, elapsedTime %v", request.NamespacedName, time.Since(start))
+		}
+	}()
+
+	// get pod and barrier
+	var pod = &v1.Pod{}
+	err = r.Get(context.TODO(), request.NamespacedName, pod)
+	if errors.IsNotFound(err) {
+		return reconcile.Result{}, nil
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
+	var barrier = &v1.ConfigMap{}
+	var barrierNamespacedName = types.NamespacedName{
+		Namespace: pod.GetNamespace(),
+		Name:      pod.Name + "-barrier",
+	}
+	err = r.Get(context.TODO(), barrierNamespacedName, barrier)
+	if errors.IsNotFound(err) {
+		barrier.Namespace = pod.GetNamespace()
+		barrier.Name = pod.Name + "-barrier"
+		barrier.OwnerReferences = append(barrier.OwnerReferences, metav1.OwnerReference{
+			APIVersion: "v1",
+			Kind:       "Pod",
+			Name:       pod.Name,
+			UID:        pod.UID,
+		})
+		temErr := r.Client.Create(context.TODO(), barrier)
+		if temErr != nil {
+			return reconcile.Result{}, temErr
+		}
+	}
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// set next starting pod
+	if !r.getConditionStatus(pod, v1.ContainersReady) {
```

Comment:
plz check the ContainersReady  condition, a pod can be not ready because of readiness gate